vars/promoteArtifacts.groovy

```diff
@@ -21,21 +21,26 @@ void call(Map args = [:]) {
     String revision = version + qualifier
     println("Revision: ${revision}")
 
-    List<String> distributionList = ['tar', 'rpm']
+    Map<String, List> distributionMap = [
+        "linux": ['tar', 'rpm'],
+        "windows": ['zip']
+    ]
+
+    println("${DISTRIBUTION_PLATFORM}" + " platform will upload these distributions to production bucket: " + distributionMap["${DISTRIBUTION_PLATFORM}"])
 
     withCredentials([string(credentialsId: 'jenkins-aws-account-public', variable: 'AWS_ACCOUNT_PUBLIC'),
         string(credentialsId: 'jenkins-artifact-bucket-name', variable: 'ARTIFACT_BUCKET_NAME'),
         string(credentialsId: 'jenkins-artifact-promotion-role', variable: 'ARTIFACT_PROMOTION_ROLE_NAME'),
         string(credentialsId: 'jenkins-aws-production-account', variable: 'AWS_ACCOUNT_ARTIFACT'),
         string(credentialsId: 'jenkins-artifact-production-bucket-name', variable: 'ARTIFACT_PRODUCTION_BUCKET_NAME')]) {
-        for (distribution in distributionList) {
+        for (distribution in distributionMap["${DISTRIBUTION_PLATFORM}"]) {
             // Must use local variable due to groovy for loop and closure scope
             // Or the stage will fixed to the last item in return when trigger new stages
             // https://web.archive.org/web/20181121065904/http://blog.freeside.co/2013/03/29/groovy-gotcha-for-loops-and-closure-scope/
             def distribution_local = distribution
             def artifactPath = "${DISTRIBUTION_JOB_NAME}/${revision}/${DISTRIBUTION_BUILD_NUMBER}/${DISTRIBUTION_PLATFORM}/${DISTRIBUTION_ARCHITECTURE}/${distribution_local}"
             def prefixPath = "${WORKSPACE}/artifacts/${distribution_local}"
-            println("S3 download ${distribution_local} artifacts before creating signatures")
+            println("S3 download ${DISTRIBUTION_BUILD_NUMBER} ${DISTRIBUTION_PLATFORM} ${DISTRIBUTION_ARCHITECTURE} ${distribution_local} ${DISTRIBUTION_JOB_NAME} ${revision} artifacts before creating signatures")
```

Comment:
Maybe we add the `BUILD_NUMBER` at the end saying `download ...${revision} artifacts from ${DISTRIBUTION_JOB_NAME} Job number ${DISTRIBUTION_BUILD_NUMBER} before...` could be more informative.
Otherwise currently the build number looks quite confusing.
e.g. `S3 download linux x64 rpm 2.0.0-rc1 artifacts from vars-build 33 before creating signatures`

Reply:
Will resolve.